### ISSUE TITLE
why did rust#96214 get tagged as regression?

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rustc-perf#1302]
Spawned off of https://github.com/rust-lang/rust/pull/96214#issuecomment-1104662390

Basically, I'm trying to understand how the PR sequence of rust-lang/rust#95379; rust-lang/rust#96020; rust-lang/rust#96214 ended up with both PR rust-lang/rust#96020 *and* PR rust-lang/rust#96214 being flagged as regressions.

From my perspective, the data next to no change between rust-lang/rust#96020 and rust-lang/rust#96214 on this benchmark, at least from eyeballing the graph:

![image](https://user-images.githubusercontent.com/173127/164365005-274a9ca7-31ea-40e9-8906-078c051d47b6.png)

(but it could be that I'm very tired and overlooking something)

**Comment by nnethercote** [comment]
Here's the changes for #96020:
![d1](https://user-images.githubusercontent.com/1940286/164371903-3c67afe6-6c28-40df-be7b-152720d0a87a.png)

Here's the changes for #96214:
![d2](https://user-images.githubusercontent.com/1940286/164371909-6763e4b5-9d52-4678-83bd-ccfe55640f64.png)

The profiles/scenarios only have partial overlap. The graph you showed above is for one profile/scenario combination where the first commit made a significant regression but the second commit didn't. So it's a case where looking at one graph is misleading, and the text on the "compare" page is more instructive.

Also, the raw data values match up like you'd expect. E.g. look at the `check full` results in both tables.

In conclusion, I think the perf bot did the right thing here.

**Comment by pnkfelix** [comment]
Okay, so the mistake I made was in focusing in on the incr-full line while looking into rust-lang/rust#96214, when I should have been looking at the incr-unchanged line, right?

Hmm: https://perf.rust-lang.org/index.html?start=2022-03-20&end=4ca19e09d302a4cbde14f9cb1bc109179dc824cd&benchmark=deeply-nested-multi&profile=opt&stat=instructions:u

and zooming in:

![image](https://user-images.githubusercontent.com/173127/164718214-7571632a-929e-4ae0-952d-9aa61a9f2793.png)


I can at least now see how rust-lang/rust#96214 got tagged.

The difference for this profile/scenario are dwarfed by the differences observed in the other profiles/scenarios. I'm still musing about whether its "right" that this got tagged as a regression, but it was a good thing to have to dig into, I guess...
